### ISSUE TITLE
Date properties not converted back into Dates

### ISSUE THREAD

**Issue by martindale** [conancat/mongoose-redis-cache#3]
When the schema has a property of type `Date`, these values are cast to Objects instead, and don't have the corresponding Date functions:

```
Object 2013-07-03T02:12:10.896Z has no method 'getFullYear'
```

**Comment by lbeschastny** [comment]
It's a common problem of `JSON` serialization.

The only way to solve it is to change serialization method. I know only one serialization library capable of parsing dates - [json-literal](https://github.com/ForbesLindesay/json-literal), but it's **ten times slower** then `JSON`.

**Comment by martindale** [comment]
Perhaps the middleware could check the types of each property on the model after deserialization, then do appropriate typecasting...

**Comment by lbeschastny** [comment]
I don't think it'll work. Certainly not for populated documents.

It may be solved with custom `parse` or with some post-processing function, though.

**Comment by conancat** [comment]
sorry guys, i have been busy with work and all and haven't been working on this project for some time. i'll look into this issue this weekend and see what i can do!

**Comment by conancat** [comment]
yes @lbeschastny is right, typecasting after parsing the JSON documents from redis are really, really slow the last time i tried using it. but maybe mongoose's typecasting performance had improved since the last time i tried it. i'll try again this weekend. :)

**Comment by lbeschastny** [comment]
I think the best solution for this problem is to allow users to specify their own serialization function.

So, setting serialization function to [json-literal](https://github.com/ForbesLindesay/json-literal) will solve all problems with dates slowing down the cache.

**Comment by martindale** [comment]
@conancat how'd it go?
